### ISSUE TITLE
HIR printing of `existential type` is broken

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust#60662]
```rust
pub existential type ServeFut: Future<Output = ()>;
```

expanded with `rustc --edition 2018 -Z unpretty=hir` yields

```rust
pub existential type ServeFut : : Future<Output = ()>;
```

we should not print that extra `:`

**Comment by Pulkit07** [comment]
Hi, I will like to work on this issue.

**Comment by oli-obk** [comment]
awesome!

Most likely all that is required is to remove https://github.com/rust-lang/rust/blob/d58cb934cb78b1f11c2fa44c469621c10a8359fd/src/librustc/hir/print.rs#L649

you can check if there's a test in `src/tests/ui` that has a `// compile-flags` comment with `-Z unpretty=hir` in it. If there is, just add the above test line to it and run tests with `./x.py test --stage 1 src/test/ui --test-args name-of-the-test --bless`.

If there is no such test, add such a test ;)

**Comment by alexreg** [comment]
I think I fixed this in one of my PRs, but not sure when it will merge, so go for it. :-)

**Comment by Pulkit07** [comment]
While adding test, I am getting following error message:

`unexpected error: '8:5: 8:43: existential types are unstable [E0658]'`

The existing test can be seen here: https://github.com/rust-lang/rust/pull/60694/files#diff-b122eeedc759d82eab86eafd6445ce8e

**Comment by oli-obk** [comment]
you need to add `#![feature(existential_types)]` to the test

**Comment by Pulkit07** [comment]
> you need to add `#![feature(existential_types)]` to the test

Added that, many thanks!
